pyfstat/mcmc_based_searches.py

```diff
@@ -998,14 +998,21 @@ def plot_chainconsumer(self, save_fig=True, label_offset=0.25, dpi=300, **kwargs
             )
             return
 
-        if "truths" in kwargs and len(kwargs["truths"]) != self.ndim:
-            logging.warning("len(Truths) != ndim, Truths will be ignored")
-            kwargs["truths"] = None
 
         samples_plt = copy.copy(self.samples)
         labels = self._get_labels(newline_units=True)
 
         samples_plt = self._scale_samples(samples_plt, self.theta_keys)
+        if "truth" in kwargs:
+            if len(kwargs["truth"]) != self.ndim:
+                logging.warning("len(Truths) != ndim, Truths will be ignored")
```

Comment:
That current check is overzealous and insufficient at the same time: it causes the `truth=signal_parameters` example to be ignored (because it contains _too many_ injection parameters), but would not catch cases where the _right number_ of _completely wrong_ keys are given. Let's to this properly:

```suggestion
            if not isinstance(kwargs["truth"], dict):
                raise ValueError("'truth' must be a dictionary.")
            missing_keys = np.setdiff1d(self.theta_keys, list(kwargs["truth"].keys()))
            if len(missing_keys) > 0:
                logging.warning(
                    "plot_chainconsumer(): Missing keys {} in 'truth' dictionary,"
                    " argument will be ignored.".format(missing_keys)
                )
```

(In keeping with numpy's general apparent disregard of pythons "one and only one way to do something" mantra, there are probably many alternatives to `setdiff1d` in this context, but this one seems to work.)

Reply:
Note: an analogus fix should probably be applied to plot_corner().

Reply:
This did the trick!